tests/test_config.py

```diff
@@ -460,3 +424,439 @@ def test_debug_override(mocker: pytest_mock.MockFixture):
     assert configFile["database"]["username"] == "fakeuser"
     assert configFile["database"]["password"] == "fakepassword"
     assert configFile["database"]["local_suffix"] == False
+
+
+def test_parse_db_string_to_dbconfig():
+    db_string = "postgresql://user:password@localhost:5432/dbname"
+    db_config = parse_db_string_to_dbconfig(db_string)
+    assert db_config["hostname"] == "localhost"
+    assert db_config["port"] == 5432
+    assert db_config["username"] == "user"
+    assert db_config["password"] == "password"
+    assert db_config["app_db_name"] == "dbname"
+
+    db_string = "postgresql://user:password@localhost:5432/dbname?connect_timeout=10&sslmode=require&sslcert=ca.pem"
+    db_config = parse_db_string_to_dbconfig(db_string)
+    assert db_config["ssl"] == True
+    assert db_config["ssl_ca"] == "ca.pem"
+    assert db_config["connectionTimeoutMillis"] == 10000
+
+    # Test unusual but valid DB strings
+    db_string = "postgresql://user:complex%23password@hostname.with.dots:5432/dbname?sslmode=require&application_name=myapp"
+    db_config = parse_db_string_to_dbconfig(db_string)
+    assert db_config["hostname"] == "hostname.with.dots"
+    assert db_config["password"] == "complex#password"  # Ensure URL decoding works
+
+    # Missing required field
+    with pytest.raises(Exception):
+        db_string = "invalid"
+        parse_db_string_to_dbconfig(db_string)
```

Comment:
We can certainly add more tests... the idea is that additional keys are ignored and we parse the ssl / connect_timeout stuff